hs_access_control/models/user.py

```diff
@@ -2241,5 +2414,1036 @@ def can_undo_share(self, **kwargs):
+
+        :param this_community: community to check
+        :return: Boolean: whether user is an owner.
+
+        Usage:
+        ------
+
+            if my_user.owns_community(g):
+                # do something that requires community ownership
+                g.description='some description'
+                g.save()
+                my_user.unshare_community_with_group(another_user,g) # e.g.
+
+        """
+        if __debug__:  # during testing only, check argument types and preconditions
+            assert isinstance(this_community, Community)
+
+        if not self.user.is_active:
+            raise PermissionDenied("Requesting user is not active")
+
+        return UserCommunityPrivilege.objects.filter(community=this_community,
+                                                     privilege=PrivilegeCodes.OWNER,
+                                                     user=self.user).exists()
+
+    def can_change_community(self, this_community):
+        """
+        Return whether a user can change this community, including the effect of resource flags.
+
+        :param this_community: community to check
+        :return: Boolean: whether user can change this community.
+
+        For communities, ownership implies change privilege but not vice versa.
+        Note that change privilege does not apply to community flags, including
+        active, shareable, discoverable, and public. Only owners can set these.
+
+        Usage:
+        ------
+
+            if my_user.can_change_community(g):
+                # do something that requires change privilege with g.
+        """
+        if __debug__:  # during testing only, check argument types and preconditions
+            assert isinstance(this_community, Community)
+
+        if not self.user.is_active:
+            raise PermissionDenied("Requesting user is not active")
+
+        if self.user.is_superuser:
+            return True
+
+        return this_community in self.edit_communities
+
+    def can_view_community(self, this_community):
+        """
+        Whether user can view this community in entirety
+
+        :param this_community: community to check
+        :return: True if user can view this resource.
+
+        Usage:
+        ------
+
+            if my_user.can_view_community(g):
+                # do something that requires viewing g.
+
+        See can_view_metadata below for the special case of discoverable resources.
+
+        Note that inferred viewers -- as determined by communities -- affect view privilege,
+        and these can view communities of which they are a community.
+        """
+        if __debug__:  # during testing only, check argument types and preconditions
+            assert isinstance(this_community, Community)
+
+        if not self.user.is_active:
+            raise PermissionDenied("Requesting user is not active")
+
+        if self.user.is_superuser:
+            return True
+
+        return self.user.is_superuser or \
```

Comment:
`self.user.is_superuser` is not needed?

Reply:
IT's the same in every routine at this level. is_superuser gives the admin account permission to do this. It is only used by CUAHSI. I think it's warranted.